regression-test/suites/nereids_rules_p0/infer_predicate/extend_infer_equal_predicate.groovy

```diff
@@ -0,0 +1,288 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+suite("extend_infer_equal_predicate") {
```

Comment:
1) add ut for non-inner join cases
2) add ut containing subquery, e.g, with constant propagation, etc.
3) add ut which outer->inner may happen.

Reply:
case: select * from t1 where t1.c1 exists (select t2.c2 from t2, t3 where t1.c2 = t2.c3 and t2.c3 = t3.c1) and t1.c2 = 1;